portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java

```diff
@@ -33,6 +33,16 @@
 @WebServlet("/delete-cmt")
 public class DeleteDataServlet extends HttpServlet {
 
+  private void deleteDataStoreComments(PreparedQuery results, DatastoreService datastore){
+    
+    while(results.countEntities() != 0){
```

Comment:
Glad to see you fixed the bug, but I'm just a little confused as to how this works. 

It doesn't look like `results` is being modified in this function... or is it? Does calling `datastore.delete` also remove the entity from `PreparedQuery results`?

Reply:
Actually, it looks like `PreparedQuery` does not cache results - so each usage of `results.countEntities()` makes a new call to the datastore (https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/PreparedQuery). Nicely done!!

Reply:
I think that this line of code ```PreparedQuery results = datastore.prepare(commentsQuery);``` ties datastore and results together.

Thus results is acting like a reference query to datastore. 

In my testing, when I printed ```results.countEntities()``` in the console it display: '2', then '0'. 2 is what was causing the bug.

Reply:
That makes sense! Can you see if it will still work if you create an empty while loop as follows:
`for (Entity entity : results.asIterable()){
        datastore.delete(entity.getKey());
  }

while(results.countEntities() != 0){
};`
And the while loop would occur after the for l
I guess in your current approach you might be calling `datastore.delete()` with the same key multiple times.